ivy/functional/ivy/nest.py

```diff
@@ -252,10 +284,31 @@ def map_nest_at_index(
     }
 
     """
+    outer_tupled = False
+    if _result is None:
+        if shallow:
+            _result = type(nest)(nest)
+        else:
+            _result = copy_nest(nest)
+        if isinstance(nest, tuple):
+            outer_tupled = True
+            _result = list(_result)
     if len(index) == 1:
-        nest[index[0]] = fn(nest[index[0]])
+        ret = fn(nest[index[0]])
+        if shallow:
+            nest[index[0]] = ret
+        _result[index[0]] = ret
     else:
-        map_nest_at_index(nest[index[0]], index[1:], fn)
+        inner_tuppled = False
+        if isinstance(_result[index[0]], tuple):
+            _result[index[0]] = list(_result[index[0]])
+            inner_tuppled = True
+        map_nest_at_index(nest[index[0]], index[1:], fn, shallow, _result[index[0]])
+        if inner_tuppled:
+            _result[index[0]] = tuple(_result[index[0]])
+    if outer_tupled:
+        _result = tuple(_result)
+    return _result
```

Comment:
I'd say a lot of the reviews for the first function make sense here as well and can enhance the code a little bit